collector/collector_test.go

```diff
@@ -1033,6 +1033,117 @@ func TestIndexesToLabels(t *testing.T) {
 	}
 }
 
+func TestConfigureTarget(t *testing.T) {
```

Comment:
Aren't you lacking positive tests for `tcp://`?

Reply:
Ah yes, it got a bit late yesterday and overlooked that.
Will add the tests shortly!

Reply:
I added more test cases now in 7212e572 covering the `tcp://` scheme.
Also added test cases for `badport` parsing using `udp://` and `tcp://` schemes as well.